### PR TITLE
Email tweaks

### PR DESCRIPTION
**Pull request by samuelcolvin**
## Change Summary

Some small and subtle tweaks to email validation. This will probably need to wait until v2 to be released.

As far as I know the following things have changed (but [`email-validator`](https://github.com/JoshData/python-email-validator)'s tests aren't that clear, so I could be wrong):
* whitespace is stripped from names, e.g. from `' Fred Smith <fred@example.com>'` the name would now be `'Fred Smith'`, before it would have been `' Fred Smith'`
* the local part can be changed in the case of some strange encodings
* the domain part can be changed in IDNA cases, e.g. `xn--80ak6aa92e.com` is changed to `аррӏе.com`

~~**Also need to add a best error message**~~ done.

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [ ] ~~Documentation reflects the changes where applicable~~
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

File: tests/test_networks.py [hunk 1]
```diff
@@ -321,6 +321,9 @@ class Model(BaseModel):
         ('foo.bar@example.com ', 'foo.bar', 'foo.bar@example.com'),
         ('foo BAR <foobar@example.com >', 'foo BAR', 'foobar@example.com'),
         ('FOO bar   <foobar@example.com> ', 'FOO bar', 'foobar@example.com'),
+        (' Whatever <foobar@example.com>', 'Whatever', 'foobar@example.com'),
+        ('Whatever < foobar@example.com>', 'Whatever', 'foobar@example.com'),
+        ('Whatever < foobar@example.com>', 'Whatever', 'foobar@example.com'),
```
Comment (graue70) [inline tests/test_networks.py:326]: These are two identical tests, aren't they?

**Review by samuelcolvin — COMMENTED** [review]
Experiment in reviews "please update".

File: tests/test_networks.py [hunk 2]
```diff
@@ -767,31 +775,36 @@ def test_address_valid(value, name, email):
...
+        ('foobar <foobar<@example.com>', None),
+        ('foobar <foobar@example.com>>', None),
+        ('foobar <<foobar<@example.com>', None),
+        ('foobar <>', None),
     ],
 )
-def test_address_invalid(value):
-    with pytest.raises(ValueError):
+def test_address_invalid(value, reason):
+    with pytest.raises(EmailError, match='value is not a valid email address: ' + (reason or '')):
```
Comment (hramezani) [inline tests/test_networks.py:807]:
```suggestion
    with pytest.raises(EmailError, match=f'value is not a valid email address: {(reason or "")}'):
```

**Review by hramezani — COMMENTED** [review]
LGTM :+1: 

I just left a small suggestion

Please update

File: tests/test_networks.py (hunk 2 above)
Comment (samuelcolvin) [reply tests/test_networks.py]: done, thanks

**Pull request merged**